src/compton/test/t2Compton.cc

```diff
@@ -0,0 +1,322 @@
+        0.95025760553927, 0.002535497503532, 0.0044495495432538, -4.5183679054704e-05, 0, 0, 0, 0,
+        0.11605523138559, -0.0025883973572953, 0.67868811344021, 0.0063593874505928,
+        // out_nonlin
+        4.6630056595372e-18, 8.5759248352589e-23, 1.1673356563462e-22, -2.6167370751723e-24, 0, 0,
+        0, 0, 1.5017646436551e-22, -3.9459638253177e-24, 9.1490085641636e-21, 5.3409650219456e-23,
+        5.484626607511e-24, -1.0596457235234e-25, 0, 0, 0, 0, 1.0314177969686e-23,
+        -2.9818990024258e-25, 4.2638762788237e-22, 8.5042714161135e-25, 1.8073497542315e-25,
+        -2.4297847629037e-27, 0, 0, 0, 0, 7.6278018408129e-25, -2.5980363364632e-26,
+        1.5794222622867e-23, 1.5887774510949e-25,
+        // in_nonlin
+        4.6615665467427e-18, 1.7873552446889e-22, 1.2663879157434e-22, -2.9982049815629e-24, 0, 0,
+        0, 0, 1.3770981199909e-22, -3.4369299797817e-24, 9.1220263142097e-21, 5.4325270040856e-23,
+        5.890752806507e-24, -1.2076068345897e-25, 0, 0, 0, 0, 9.5432783203125e-24,
+        -2.6517064022462e-25, 4.2104110096026e-22, 1.0546962404339e-24, 1.9308590430364e-25,
+        -2.8485633209286e-27, 0, 0, 0, 0, 7.0910968932275e-25, -2.3423156364316e-26,
+        1.4873126610905e-23, 1.8374675563093e-25};
+
+    ut.check(std::equal(flat_interp_data.begin(), flat_interp_data.end(), flat_interp_gold.begin(),
+                        [tol](double a, double b) -> bool { return soft_equiv(a, b, tol); }),
+             "checked data interpolation");
+  }
+
+  if (ut.numFails == 0)
+    std::cout << "\nCorrectly read multigroup data points!" << std::endl;
+
+  if (ut.numFails == 0) {
+    PASSMSG("Successfully linked Draco against CSK_generator.");
+  } else {
+    FAILMSG("Did not successfully link Draco against CSK_generator.");
+  }
+}
+
+//------------------------------------------------------------------------------------------------//
+//!  Tests Compton's error-handling on a non-existent file.
+void compton_fail_test(rtt_dsxx::UnitTest &ut) {
+  std::cout << "\n---------------------------------------------------------\n"
+            << "    Test Compton bad file handling    \n"
+            << "---------------------------------------------------------\n";
+  // open a small mg opacity file:
+  std::string filename = ut.getTestSourcePath() + "non_existent.compton";
+  std::cout << "Testing with a non-existent file...\n" << std::endl;
+  std::unique_ptr<rtt_compton::Compton> compton_test;
+
+  bool caught = false;
+  try {
+    compton_test.reset(new rtt_compton::Compton(filename));
+  } catch (rtt_dsxx::assertion &asrt) {
+    std::cout << "Draco exception thrown: " << asrt.what() << std::endl;
+    // We successfully caught the bad file!
+    caught = true;
+  } catch (const int & /*asrt*/) {
+    std::cout << "CSK exception thrown. " << std::endl;
+    // We successfully caught the bad file!
+    caught = true;
+  }
```

Comment:
You might also want the `catch(...)` case. You can omit it here and just let the top level catch work its magic.

Reply:
Is it just replace the catch statement with " catch(...)" or does something special need to replace the ellipses. I'm not overly familiar with this language feature or exceptions in general

Reply:
> Is it just replace the catch statement with " catch(...)" or does something special need to replace the ellipses. I'm not overly familiar with this language feature or exceptions in general

Since this is is a test, I think `catch(...)` is appropriate.  You can fine tune the error message or recovery behavior with a more detailed `catch(arg)`, if there is need to do so.